src/main/java/com/google/sps/data/Keywords.java

```diff
@@ -0,0 +1,132 @@
+package com.google.sps.data;
+
+import java.io.IOException;
+import com.google.cloud.language.v1.Document;
+import com.google.cloud.language.v1.LanguageServiceClient;
+import com.google.cloud.language.v1.AnalyzeEntitiesRequest;
+import com.google.cloud.language.v1.AnalyzeEntitiesResponse;
+import com.google.cloud.language.v1.EncodingType;
+import com.google.cloud.vision.v1.EntityAnnotation;
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.EntityNotFoundException;
+import com.google.appengine.api.datastore.FetchOptions;
+import com.google.appengine.api.datastore.Key;
+import com.google.appengine.api.datastore.KeyFactory;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query.SortDirection;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+import java.util.HashSet;
+import java.util.Set;
+import java.util.TreeSet;
+
+/**
+ * Provide methods for the storage, extraction, and retrieval of keywords.
+ */
+public final class Keywords {
+
+  private static final int MAX_NUM_KEYWORDS = 10;
+
+  /**
+   * @return a list of the 10 most salient keywords
+   */
+  public static List<String> getKeywords(String key) {
+    try {
+      Key datastoreKey = KeyFactory.stringToKey(key);
+      DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+      Entity result = datastore.get(datastoreKey);
+    
+      Collection<String> keywordCollection = (Collection<String>) result.getProperty("keywords");
+      List<String> keywordList = new ArrayList<>();
+      int index = 0;
+      for (String keyword : keywordCollection) {
+        keywordList.add(keyword);
+    
+        index++;
+        if (index >= MAX_NUM_KEYWORDS) {
+          break;
+        }
+      }
+      return keywordList;
+    } catch (EntityNotFoundException ex) {
+      return new ArrayList<>();
+    }
+  }
+
+  /**
+   * Adds the labels from the analysis of an image into the datastore.
+   * @return the key pointing to the newly added datastore entity.
+   */
+  public static String addKeywords(List<EntityAnnotation> blobAnalysis) throws IOException {
+    String labels = "";
+    for (EntityAnnotation label : blobAnalysis) {
+      labels += label.getDescription() + ", ";
```

Comment:
The way this is currently written will mean that labels will have a trailing ", ".  Is that intended?  If not, I would suggest making a list of label.getDescription() for each label and then calling String.join() on that list.

Reply:
The trailing ", " does not affect the extraction of salience from the labels string, so although not really intended, it also doesn't affect the functionality of anything else in the class.

Reply:
Hmm, I guess that's true but I think it is still better to use String.join here since what you really _want_ is a comma separated string.  You can just create a List<String> inside the for loop and then join it at the end.